puppet/modules/jenkins_job_builder/files/theforeman.org/unmanaged_jobs.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+
+require 'net/http'
+require 'json'
+
+def managed?(job)
+  job['description']&.include?('<!-- Managed by Jenkins Job Builder -->')
+end
+
+def get_jenkins_url_from_ini(ini_file)
+  lines = File.open(ini_file, "r") do |file|
+    file.readlines
+  end
+
+  url_line = lines.select { |line| line.include?('url=') }.first
+  url = url_line.gsub(/url=(.*\/\/)?/, '').chomp
+  url
+end
+
+def find_unmanaged_jobs_from_payload(json_payload)
+  payload = JSON.parse(json_payload)
+
+  um_jobs = payload['jobs'].select { |job| !managed?(job) }
```

Comment:
I think you could also do `payload['jobs'].reject { |job| managed?(job) }` and leave out the variable assignment.

Reply:
Good idea!